File: src/ObjectManager/Object.Tes/FilePacks/Records/345-GMST.Game Setting.cs
using OA.Core;

namespace OA.Tes.FilePacks.Records
{
    public class GMSTRecord : Record, IHaveEDID
    {
        public override string ToString() => $"GMST: {EDID.Value}";
        public STRVField EDID { get; set; } // Editor ID
        public DATVField DATA; // Data

        public override bool CreateField(UnityBinaryReader r, GameFormatId format, string type, int dataSize)
        {
            if (format == GameFormatId.TES3)
                switch (type)
                {
                    case "NAME": EDID = new STRVField(r, dataSize); return true;
                    case "STRV": DATA = new DATVField(r, dataSize, 's'); return true;
                    case "INTV": DATA = new DATVField(r, dataSize, 'i'); return true;
                    case "FLTV": DATA = new DATVField(r, dataSize, 'f'); return true;
                    default: return false;
                }
            switch (type)
            {
                case "EDID": EDID = new STRVField(r, dataSize); return true;
                case "DATA": DATA = new DATVField(r, dataSize, EDID.Value[0]); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/345-GMST.Game Setting.swift
//
//  GMSTRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class GMSTRecord: Record, IHaveEDID {
    public override var description: String { return "GMST: \(EDID)" }
    public var EDID: STRVField = STRVField_empty // Editor ID
    public var DATA: DATVField! // Data

    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        if format == .TES3 {
            switch type {
            case "NAME": EDID = r.readSTRV(dataSize)
            case "STRV": DATA = r.readDATV(dataSize, type: "s")
            case "INTV": DATA = r.readDATV(dataSize, type: "i")
            case "FLTV": DATA = r.readDATV(dataSize, type: "f")
            default: return false
            }
            return true
        }
        switch (type)
        {
        case "EDID": EDID = r.readSTRV(dataSize)
        case "DATA": DATA = r.readDATV(dataSize, type: EDID.first!)
        default: return false
        }
        return true
    }
}
